### ISSUE TITLE
Name 'typing.io.IO' is not defined

### ISSUE THREAD

**Issue by FichteFoll** [python/mypy#1462]
The [Python 3.5 docs](https://docs.python.org/3/library/typing.html?highlight=typing.io#typing.io) list a `typing.io` *something* ("wrapper namespace" according to docs, class according to interpreter), which you can access with `IO` to get an IO type, or for example `TextIO`. Using this in an interpreter works fine, but mypy complains about this.

```py
>>> typing.io.IO[str]
typing.IO[str]
>>> typing.io.TextIO
typing.TextIO[str]
```
```
prettify.py:40: error: Name 'typing.io.IO' is not defined
```

I understand (now) that `typing.IO` is also possible, but the docs are *very* vague about that being a possibility, which is why I didn't even try it. Docs aside, since this works in the interpreter I think it should also work in mypy, which is why I'm reporting it.

**Comment by gvanrossum** [comment]
Right, it's meant as a namespace, it seems mypy doesn't understand this
yet. Bug.

--Guido (mobile)

**Comment by gvanrossum** [comment]
We added this to PEP 484 and typing.py in a relatively late stage, as a preferred way to separate support for specific modules from support for fundamentals (including containers). I guess mypy never implemented it. Maybe you can also submit a doc bug for Python 3.5?
